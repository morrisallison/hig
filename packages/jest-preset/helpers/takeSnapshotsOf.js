const React = require("react");
const renderer = require("react-test-renderer");

/**
 * @typedef {import("react-test-renderer").TestRendererOptions} Options
 */

/**
 * @typedef {Object} Case
 * @property {string} desc
 * @property {Object} [props]
 * @property {import("react").ReactElement} [subject]
 */

/** @type {Options} */
const defaultOptions = {
  createNodeMock(element) {
    return document.createElement(element.type);
  }
};

/**
 * @param {import("react").Component|null} Component
 * @param {Case[]} cases
 * @param {Options} [options]
 */
module.exports = function takeSnapshotsOf(Component, cases, options) {
  cases.forEach(({ desc, props, subject }) => {
    it(desc, () => {
      const rendererOptions = Object.assign({}, defaultOptions, options);
      const vDom = subject || React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
