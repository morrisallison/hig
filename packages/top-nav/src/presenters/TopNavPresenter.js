import React from "react";
import PropTypes from "prop-types";

import "./TopNavPresenter.scss";

export default function TopNav({ leftActions, rightActions, logo }) {
  return (
    <div className="hig__top-nav">
      {leftActions}
      <div className="hig__top-nav__logo-wrapper">{logo}</div>
      <div role="presentation" aria-hidden className="hig__top-nav__spacer" />
      {rightActions}
    </div>
  );
}

TopNav.propTypes = {
  /** Actions to render to the left of the logo */
  leftActions: PropTypes.node,
  /** Actions to render on the right of the bar */
  rightActions: PropTypes.node,
  /** Content to render as the logo */
  logo: PropTypes.node
};
