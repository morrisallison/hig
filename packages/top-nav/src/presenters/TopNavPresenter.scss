@import "~@hig/styles/mixins/base.scss";
@import "~@hig/styles/mixins/colors.scss";
@import "~@hig/styles/mixins/shadows.scss";

.hig__top-nav {
  @import "../tokens.scss";

  @include reset;
  @include shadow("dropshadow");

  display: flex;
  position: relative;
  height: $top-nav-height;
  min-height: $top-nav-height;
  padding-left: 7px;
  justify-content: space-between;
  align-items: stretch;
  z-index: $top-nav-layer;
  background: color(hig-white);
}

.hig__top-nav__item {
  display: flex;
  position: relative;
  flex-direction: row;
  align-items: stretch;
}

.hig__top-nav__spacer {
  flex: 1 1 0;
}

.hig__top-nav__logo-wrapper {
  display: flex;
  padding: 0 12px 0 5px;
  align-items: center;
  line-height: 0;
}
