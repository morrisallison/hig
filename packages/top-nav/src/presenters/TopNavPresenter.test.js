import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import TopNavPresenter from "./TopNavPresenter";

describe("top-nav/presenters/TopNavPresenter", () => {
  takeSnapshotsOf(TopNavPresenter, [
    {
      description: "renders with no props",
      props: {}
    },
    {
      description: "renders with actions and logo",
      props: {
        leftActions: <div className="leftActions" />,
        rightActions: <div className="rightActions" />,
        logo: <div className="logo" />
      }
    }
  ]);
});
