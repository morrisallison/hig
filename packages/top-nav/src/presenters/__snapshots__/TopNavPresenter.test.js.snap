// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-nav/presenters/TopNavPresenter  1`] = `
<div
  className="hig__top-nav"
>
  <div
    className="hig__top-nav__logo-wrapper"
  />
  <div
    aria-hidden={true}
    className="hig__top-nav__spacer"
    role="presentation"
  />
</div>
`;

exports[`top-nav/presenters/TopNavPresenter  2`] = `
<div
  className="hig__top-nav"
>
  <div
    className="leftActions"
  />
  <div
    className="hig__top-nav__logo-wrapper"
  >
    <div
      className="logo"
    />
  </div>
  <div
    aria-hidden={true}
    className="hig__top-nav__spacer"
    role="presentation"
  />
  <div
    className="rightActions"
  />
</div>
`;
